packages/grafana-ui/src/components/Forms/InlineField.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { HTMLProps, FC } from 'react';
+import { cx, css } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { useTheme } from '../../themes';
+import { getInlineLabelStyles, InlineLabel } from './InlineLabel';
+import { PopoverContent } from '../Tooltip/Tooltip';
+
+export interface Props extends Omit<HTMLProps<HTMLDivElement>, 'className' | 'css' | 'label'> {
+  /** Form input element, i.e Input or Switch */
+  children: React.ReactElement;
+  /** Label for the field. If it's html elements and not a string */
+  label?: React.ReactNode;
+  /** Content for the label's tooltip */
+  tooltip?: PopoverContent;
+  /** Custom width for the label */
+  labelWidth?: number | 'auto';
+  /** Indicates if field is in invalid state */
+  invalid?: boolean;
+  /** Indicates if field is in loading state */
+  loading?: boolean;
+  /** Indicates if field is disabled */
+  disabled?: boolean;
+  /** Custom styles for the field */
+  className?: string;
+  /** Make the field's child to fill the width of the row. Equivalent to setting `flex-grow:1` on the field */
+  grow?: boolean;
+  /** A toggle to apply query keyword styling to the label */
+  isKeyword?: boolean;
+  /** Fill the remaining width of the row with the label's background */
+  fill?: boolean;
+}
+
+export const InlineField: FC<Props> = ({
```

Comment:
Shouldn't this extend [FieldProps](https://github.com/grafana/grafana/blob/416111c5f6afb32c4036724d7f87fa7a1102055e/packages/grafana-ui/src/components/Forms/Field.tsx#L8)? I think the HTMLProps extension should belong to the FieldProps too, then you could use that type as a base for InlineField properties